src/policy/mod.rs

```diff
@@ -492,5 +494,32 @@ mod tests {
             let expected_descriptor = Descriptor::new_tr("E".to_string(), Some(tree)).unwrap();
             assert_eq!(descriptor, expected_descriptor);
         }
+
+        // private and optimized compilation for a given policy
+        {
+            let policy = policy_str!(
+                "thresh(1,or(1@pk(A),1@pk(B)),or(1@pk(C),1@or(1@and(pk(E),pk(F)),1@pk(D))))"
+            );
+            let priv_desc = policy
+                .clone()
+                .compile_tr_private(Some(unspendable_key.clone()))
+                .unwrap();
+            let priv_expected_desc = Descriptor::Tr(
+                Tr::<String>::from_str("tr(A,{{and_v(v:pk(E),pk(F)),pk(D)},{pk(C),pk(B)}})")
+                    .unwrap(),
+            );
+
+            assert_eq!(priv_desc, priv_expected_desc);
+
+            let opt_desc = policy
+                .clone()
+                .compile_tr(Some(unspendable_key.clone()))
+                .unwrap();
+            let opt_expected_desc = Descriptor::Tr(
+                Tr::<String>::from_str("tr(A,{{pk(D),pk(C)},{pk(B),and_v(v:pk(E),pk(F))}})")
```

Comment:
Did we succeed in finding any policy where these differs?

Reply:
We did find policies where the TapTree leaf nodes re-arranged, but none yet where we noticed policy-merging. One of those were:
Policy: `thresh(1,or(1@pk(A),1@pk(B)),or(1@pk(C),1@or(1@and(pk(E),pk(F)),1@pk(D))))`
Private compilation: `tr(A,{{and_v(v:pk(E),pk(F)),pk(D)},{pk(C),pk(B)}})#y7cmc3xl`
Default compilation: `tr(A,{{pk(D),pk(C)},{pk(B),and_v(v:pk(E),pk(F))}})#vv7casf3`